preprocessors/line-charts/charts.py

```diff
@@ -0,0 +1,132 @@
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# and our Additional Terms along with this program.
+# If not, see
+# <https://github.com/Shared-Reality-Lab/IMAGE-server/blob/main/LICENSE>.
+
+
+import json
+import time
+import logging
+import jsonschema
+from flask import Flask, request, jsonify
+
+from charts_utils import getLowerPointsOnLeft, getHigherPointsOnLeft
+from charts_utils import getLowerPointsOnRight, getHigherPointsOnRight
+
+app = Flask(__name__)
+
+
+@app.route('/preprocessor', methods=['POST', 'GET'])
+def get_chart_info():
+    """
+    The preprocessor currently handles only single-line charts,
+    functionality to be extended later
+    """
+    logging.debug("Received request")
+    # Load schemas
+    with open('./schemas/preprocessors/line-charts.schema.json') as jsonfile:
+        data_schema = json.load(jsonfile)
+    with open('./schemas/preprocessor-response.schema.json') as jsonfile:
+        schema = json.load(jsonfile)
+    with open('./schemas/definitions.json') as jsonfile:
+        definition_schema = json.load(jsonfile)
+    schema_store = {
+        data_schema['$id']: data_schema,
+        schema['$id']: schema,
+        definition_schema['$id']: definition_schema
+    }
+    content = request.get_json()
+
+    # Check if request is for a chart
+    if 'highChartsData' not in content:
+        logging.info("Not a highcharts charts request. Skipping...")
+        return "", 204
+
+    with open('./schemas/request.schema.json') as jsonfile:
+        request_schema = json.load(jsonfile)
+    # Validate incoming request
+    resolver = jsonschema.RefResolver.from_schema(
+        request_schema, store=schema_store)
+    try:
+        validator = jsonschema.Draft7Validator(
+            request_schema,
+            resolver=resolver
+        )
+        validator.validate(content)
+    except jsonschema.exceptions.ValidationError as error:
+        logging.error(error)
+        return jsonify("Invalid Request JSON format"), 400
+    # Use response schema to validate response
+    resolver = jsonschema.RefResolver.from_schema(
+        schema, store=schema_store)
+
+    name = 'ca.mcgill.a11y.image.preprocessor.lineChart'
+    request_uuid = content['request_uuid']
+    timestamp = int(time.time())
+
+    series_object = content['highChartsData']['series'][0]
+    series_data = series_object['data']
+
+    for index, point in enumerate(series_data):
+        point['lowerPointsOnLeft'] = getLowerPointsOnLeft(
```

Comment:
I'm still worried about the efficiency here, but since this is important I'll just ask @jeffbl for his input as well.